pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_status.go

```diff
@@ -0,0 +1,106 @@
+package gatewayapi
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	kube_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kube_schema "k8s.io/apimachinery/pkg/runtime/schema"
+	kube_types "k8s.io/apimachinery/pkg/types"
+	kube_client "sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+func (r *HTTPRouteReconciler) findReference(gk kube_schema.GroupKind) (kube_client.Object, error) {
+	versions := r.Scheme.VersionsForGroupKind(
+		kube_schema.GroupKind{Group: gk.Group, Kind: gk.Kind},
+	)
+
+	if len(versions) == 0 {
+		return nil, nil
+	}
+
+	gvk := kube_schema.GroupVersionKind{
+		Group:   gk.Group,
+		Kind:    gk.Kind,
+		Version: versions[0].Version,
+	}
+
+	rawObj, err := r.Scheme.New(gvk)
+	if err != nil {
+		return nil, err
+	}
+
+	obj, ok := rawObj.(kube_client.Object)
+	if !ok {
+		return nil, nil
+	}
+
+	return obj, nil
+}
+
+func (r *HTTPRouteReconciler) resolvedRefCondition(
+	ctx context.Context, route kube_client.Object, ref gatewayapi.ParentRef,
+) (kube_meta.Condition, error) {
+	namespace := route.GetNamespace()
+	if ns := ref.Namespace; ns != nil {
+		namespace = string(*ns)
+	}
```

Comment:
FWIW controllers are supposed to use `ReferencePolicy` when crossing namespaces. That's potentially a large topic that is worth breaking out into on or more separate issues.

Reply:
I just realized that this also applies to `gapiToKumaRoute`

Reply:
I have this as a point in the compliance issue, I'll break it out into a separate issue.